sat/options/options.go

```diff
@@ -0,0 +1,76 @@
+package options
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"github.com/sethvargo/go-envconfig"
+)
+
+type Options struct {
+	EnvToken string   `env:"SAT_ENV_TOKEN"`
+	Port     port     `env:"SAT_HTTP_PORT"`
+	ProcUUID procUUID `env:"SAT_UUID"`
+
+	ControlPlane *ControlPlane `env:",noinit"`
+	Ident        *Ident        `env:",noinit"`
+	Version      *Version      `env:",noinit"`
+	TracerConfig TracerConfig  `env:",prefix=SAT_TRACER_"`
+}
+
+// ControlPlane is a struct, so we can use a pointer, so we can check whether it's been set in config. If set, it holds
+// address where the control plane is located so Sat can pull runnables from there.
+type ControlPlane struct {
+	Address string `env:"SAT_CONTROL_PLANE"`
+}
+
+// Ident holds the runnable ident. It's a struct, so we can optionally have it set to nil. Config uses it.
+type Ident struct {
```

Comment:
Why do these need to be structs instead of just strings in the `Options` struct?

Reply:
> It's a struct, so we can optionally have it set to nil.

If it's a string, a non-existing environment variable will end up populating the field as a valid emptystring. That's no different than having the environment variable set to the value of empty string.

Here we need to make a distinction between
* does not exist
* exists and is emptystring
* exists and is not emptystring

A string type would only make a difference between these two:
* either doesn't exist, or exists, but is empty string
* exists and is not empty string